### ISSUE TITLE
.IsMenuCurrent returns false when menu configured via config file vs. front matter 

### ISSUE THREAD

**Issue by chalin** [gohugoio/hugo#9150]
I'll use the [docsy-example] to illustrate this bug which appears in Hugo 0.86+ (including 0.89.2):

 1. Clone or download [docsy-example], then run `npm install`
 2. Add the following to `config.toml`:
     > [[menu.main]]
	name = "Blog"
	url = "/blog/"
	weight = 30
 3. Remove the `menu` entry data from the front matter of `content/en/blog/_index.md`
 4. Run `hugo serve` and visit http://localhost:1313/
 5. Select the **Blog** entry from the navbar

### What do I expect to see

The **Blog** entry in the navbar should be bold (because it is active), which is what I you'll when using Hugo 0.85 or earlier.

### What do I see instead?

Using Hugo 0.86 or later, the Blog entry isn't bold because it doesn't have the `active` class associated with it

### More info

It seems that [.IsMenuCurrent] returns false for the Blog page when using Hugo 0.86+. Here is the line where this function is used in docsy, https://github.com/google/docsy/blob/1b167606bd62d52097357d7cd533f7415a77d17d/layouts/partials/navbar.html#L11

[docsy-example]: https://github.com/google/docsy-example
[.IsMenuCurrent]: https://gohugo.io/functions/ismenucurrent/

### Related

- https://github.com/etcd-io/website/issues/443
- https://github.com/notaryproject/notaryproject.dev/issues/18
- And on other sites since upgrading Hugo

/cc @austinlparker @nate-double-u @celestehorgan

**Comment by jmooring** [comment]
@chalin 

This is actually a duplicate of #3459 which was closed by stalebot (no action) about four years ago. It's not a new problem.

But there's good news. In v0.86.0 a new menu property (`pageref`) was created. Use `pageref` instead of `url` when defining menus via site configuration. See <https://gohugo.io/variables/menus/#menu-entry-variables>.

In short, instead of doing this:

```toml
[[menu.main]]
name = 'About'
url = '/about/'
weight = 1
```

Do this:

```toml
[[menu.main]]
name = 'About'
pageref = '/about'
weight = 1
```

Obviously you will still use `url` when creating menu links to external sites.

Here's a minimal reproducible example:

```bash
git clone --single-branch -b hugo-github-issue-9150 https://github.com/jmooring/hugo-testing hugo-github-issue-9150
cd hugo-github-issue-9150
hugo server
```

When you run the site you will see three menus at the top of each page:
- One defined in front matter
- One defined in site configuration using the `pageref` property
- One defined in site configuration using the `url` property

The first two mark the active class correctly for all menu entries.
The last one marks the active class correctly only for the home page.

**Comment by chalin** [comment]
@jmooring - thank you for the detailed feedback and the reproducible example (that is an amazing resource base to have)! ✨

**Comment by chalin** [comment]
Works like a charm, thanks!

**Comment by jmooring** [comment]
@chalin I am re-opening this. The original issue was closed by stale bot, and the underlying problem is not resolved, even if the fix is just to document the limitation.

**Comment by tuthugo** [comment]
Happens to me with translated pages where `pageref` works (with URL defined) but does not work in multilingual site https://discourse.gohugo.io/t/active-class-for-translated-menus/39415/13

**Comment by jmooring** [comment]
This is resolved by using pageRef (introduced in v0.86.0) instead of url when defining menu entries in site configuration. This is now a documentation issue, to be addressed shortly with https://github.com/gohugoio/hugoDocs/issues/1594.
